File: equipo1_ej2.py
def detectar_tipo_dato(entrada):
    try:
        float(entrada)
        return "numérico"
    except ValueError:
        if entrada.lower() in ['true', 'false']:
            return "booleano"
        else:
            return "cadena"

# Ejemplo de uso
dato = input("Ingrese un dato: ")
tipo = detectar_tipo_dato(dato)
print("El dato ingresado es de tipo:", tipo)

File: Equipo1_ej2.java
//JAVA
import java.util.Scanner;

public class Equipo1_ej2 {
    public static String detectarTipoDato(String entrada) {
        try {
            Double.parseDouble(entrada);
            return "numérico";
        } catch (NumberFormatException e) {
            if (entrada.equalsIgnoreCase("true") || entrada.equalsIgnoreCase("false")) {
                return "booleano";
            } else {
                return "cadena";
            }
        }
    }
}